datafusion/optimizer/src/optimizer.rs

```diff
@@ -304,43 +284,25 @@ impl TreeNodeRewriter for Rewriter<'_> {
 
     fn f_down(&mut self, node: LogicalPlan) -> Result<Transformed<LogicalPlan>> {
         if self.apply_order == ApplyOrder::TopDown {
-            optimize_plan_node(node, self.rule, self.config)
+            {
```

Comment:
minor is that it is strange to have this second level of nesting braces -- we can remove it as a follow on PR I think